### PR TITLE
GH-91432: Implement the FOR_ITER_SET specialization

### PR DESCRIPTION
**Pull request by corona10**
## Microbenchmark
result: Mean +- std dev: [base] 662 ns +- 5 ns -> [FOR_ITER_SET] 616 ns +- 2 ns: 1.07x faster

`````python
import pyperf

runner = pyperf.Runner()
runner.timeit(name="set iter",
              stmt="""
for e in s:
    pass
""",
              setup = """
s = set(range(100))
"""
              )

`````

## Leak test
`````
➜  cpython git:(gh-91432) ✗ ./python.exe -m test test_set -R 3:3
Raised RLIMIT_NOFILE: 256 -> 1024
0:00:00 load avg: 7.62 Run tests sequentially
0:00:00 load avg: 7.62 [1/1] test_set
beginning 6 repetitions
123456
......

== Tests result: SUCCESS ==

1 test OK.

Total duration: 23.3 sec
Tests result: SUCCESS
``````

I just follow @sweeneyde's work.
(Following the faster CPython project is kind of my daily hobby ;) Sorry if you already worked on your local branch.. )
According to [his stat](https://gist.github.com/sweeneyde/e1a7e98890e75c9355f99067b03ee37b), set iteration (3.8%) would be worth optimizing as same as tuple iteration (2.8%)
This PR would be decided to be merged after https://github.com/python/cpython/pull/94096 is merged.

### CODE REVIEW

**Comment by markshannon** [comment]
Is iteration over sets common enough to justify this?
The [stats on pyperformance](https://github.com/faster-cpython/ideas/blob/main/stats.md) suggest it is only ~3% of `FOR_ITER` executions.

Our main motivation for specializing `FOR_ITER` is to allow us to iterate over generators without making C calls and without slowing down iterating over sequences.

**Comment by markshannon** [comment]
The change in stats is presumably from merging https://github.com/python/cpython/pull/91713

I'll run pyperformance on this PR, but its in the queue behind https://github.com/python/cpython/pull/94109, so might take a while...

File: Objects/setobject.c [hunk 1]
```diff
@@ -791,81 +782,72 @@ static PyMethodDef setiter_methods[] = {
...
+}
+
+static PyObject *setiter_iternext(_PySetIterObject *si) {
+    PyObject *stack[1];
+    int err = _PySetIter_GetNext(si, stack);
+    if (err <= 0) {
+        return NULL;
+    }
+    return stack[0];
 }
```
Comment (corona10) [inline Objects/setobject.c:832]:
```suggestion
static PyObject *setiter_iternext(_PySetIterObject *si) {
    PyObject *key;
    if (_PySetIter_GetNext(si, &key) <= 0) {
        return NULL;
    }
    return key;
}
```

might be better

**Comment by corona10** [comment]
@markshannon agree Let's close the PR!

**Pull request closed without merging**